packages/playwright-core/src/server/network.ts

```diff
@@ -97,15 +98,17 @@ export class Request extends SdkObject {
   readonly _headers: types.HeadersArray;
   private _headersMap = new Map<string, string>();
   private _rawRequestHeadersPromise: ManualPromise<types.HeadersArray> | undefined;
-  private _frame: frames.Frame;
+  private _frame: frames.Frame | null = null;
+  readonly _context: contexts.BrowserContext;
   private _waitForResponsePromise = new ManualPromise<Response | null>();
   _responseEndTiming = -1;
   readonly responseSize: ResponseSize = { encodedBodySize: 0, transferSize: 0, responseHeadersSize: 0 };
 
-  constructor(frame: frames.Frame, redirectedFrom: Request | null, documentId: string | undefined,
+  constructor(context: contexts.BrowserContext, frame: frames.Frame | null, redirectedFrom: Request | null, documentId: string | undefined,
     url: string, resourceType: string, method: string, postData: Buffer | null, headers: types.HeadersArray) {
-    super(frame, 'request');
+    super(context, 'request');
```

Comment:
This should be `super(frame || context, 'request')` to preserve attribution to the page.